packages/desktop-client/src/components/reports/spreadsheets/grouped-spreadsheet.ts

```diff
@@ -0,0 +1,152 @@
+import { runQuery } from 'loot-core/src/client/query-helpers';
+import { send } from 'loot-core/src/platform/client/fetch';
+import * as monthUtils from 'loot-core/src/shared/months';
+import { integerToAmount } from 'loot-core/src/shared/util';
+import {
+  type CategoryEntity,
+  type CategoryGroupEntity,
+} from 'loot-core/src/types/models';
+
+import {
+  categoryLists,
+  type UncategorizedEntity,
+  type UncategorizedGroupEntity,
+} from '../ReportOptions';
+
+import { type createSpreadsheetProps } from './default-spreadsheet';
+import filterHiddenItems from './filterHiddenItems';
+import makeQuery from './makeQuery';
+import recalculate from './recalculate';
+
+function createGroupedSpreadsheet({
+  start,
+  end,
+  categories,
+  selectedCategories,
+  conditions = [],
+  conditionsOp,
+  hidden,
+  uncat,
+}: createSpreadsheetProps) {
+  let catList: CategoryEntity[] | UncategorizedEntity[];
+  let catGroup: CategoryGroupEntity[] | UncategorizedGroupEntity[];
+
+  [catList, catGroup] = categoryLists({ hidden, uncat, categories });
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏same problem here